fastlane.gemspec

```diff
@@ -56,6 +56,16 @@ Gem::Specification.new do |spec|
   # spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = Dir["*/lib"]
 
+  # need to lock under 0.15 using less than Ruby 2.5 to prevent install issues when using 'gem install'
+  # 'gem install' does not respect Ruby versions and would try installing 0.15 on Ruby 2.4 or less
+  # signet - https://github.com/googleapis/signet/commit/bd6fe87948f8fc7702720dae651e82f4fd348b5d
+  # googleauth - https://github.com/googleapis/google-auth-library-ruby/commit/6644806ab47cea6d08e1901c2ed808e53a579bc3
+  if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.5')
+    spec.add_dependency('signet', '<= 0.14.1')
+    spec.add_dependency('googleauth', '<= 0.15.1')
+    STDERR.puts("WARNING: Locking to a potentially insecure version of 'signet' and 'googleauth' because you are using a version of Ruby which is marked as End-Of-Life. Please upgrade your Ruby installation to 2.5 or later")
+  end
```

Comment:
This is one of the known issues in RubyGem or rubygem.org but this branching based on `RUBY_VERSION` doesn't effectively work. Even worse this can cause unexpected result when releasing a new version of the gem.

* https://github.com/kamui/retriable/issues/79
* https://github.com/rubocop/ruby-style-guide/issues/763

So I would suggest doing followings

* Don't lock them in `gemspec` and let users choose their dependencies
* Move this logic in `Gemfile` in order to make sure fastlane developers/contributors can install proper dependencies
* Announce `gem install fastlane` or `gem "fastlane"` in `Gemfile` starts installing new dependencies that don't support Ruby 2.4 so you need to either
   * Lock `signet` and `googleauth` in your `Gemfile`  with `gem 'signet',  '<= 0.14.1'` and `gem 'googleauth', '<= 0.15.1'`
   *  Install old dependencies with `gem install signet -v 0.14.1` and `gem install googleauth -v 0.14` before running `gem install fastlane` or `gem update fastlane`

We can fix CI in the above way (installing old ones first) but I think it's fine just to bump the used Ruby version for that checks as Ruby 2.4. will be deprecated near future anyway. If we want to make sure everyone's installation works we should prepare a different kind of checks for CI.